superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -166,14 +194,126 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOverwriteDatasetOption(data, option) {
+    this.setState({ datasetToOverwrite: option })
+  }
+
+  handleOverwriteDataset() {
+    const { sql, results } = this.props.query;
+    const { datasetToOverwrite } = this.state
+
+    // HACK: to clear the columns in the previous dataset and update
+    // it with the new selected columns from the query
+    // Todo: move this to actions file
+    SupersetClient.put({
+      endpoint: `api/v1/dataset/${datasetToOverwrite.dataSetId}`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        columns: [],
+      }),
+    })
+      .then(d => {
+        console.log(d);
+      })
+      .catch(err => console.log(err));
+
+    SupersetClient.put({
+      endpoint: `api/v1/dataset/${datasetToOverwrite.dataSetId}`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        sql,
+        columns: results.selected_columns.map(d => ({column_name: d.name}))
+      }),
+    }).then(d => {
+        console.log(d);
+      })
+      .catch(err => console.log(err));
+  }
+
+  handleSaveInDataset() {
+    // if user wants to overwrite a dataset we need to prompt them
+    if (this.state.saveDatasetRadioBtnState === 2) {
+      this.setState({overwriteDataSet: true})
+      return;
+    }
+
+    const { schema, sql, dbId, templateParams } = this.props.query;
+
+    let selectedColumns: any[];
+    if (
+      this.props.query &&
+      this.props.query.results &&
+      this.props.query.results.selected_columns
```

Comment:
if you want to use optional chaining: `this.props.query?.results?.selected_columns`